specification/communication/data-plane/Rooms/preview/2022-02-01-preview/communicationservicesrooms.json

```diff
@@ -0,0 +1,707 @@
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "removeParticipantsRequest",
+            "description": "Participants in a room to be removed.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RemoveParticipantsRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The participants were successfully deleted. The action returns an updated `Room`.",
+            "schema": {
+              "$ref": "#/definitions/RoomModel"
+            }
+          },
+          "default": {
+            "description": "Error response",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "x-ms-client-name": "ErrorCode",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Remove participant": {
+            "$ref": "./examples/Rooms_RemoveParticipants.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CommunicationErrorResponse": {
+      "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationErrorResponse"
+    },
+    "CreateRoomRequest": {
+      "description": "Request payload for creating new room.",
+      "type": "object",
+      "properties": {
+        "validFrom": {
+          "format": "date-time",
+          "description": "The timestamp from when the room is open for joining. The timestamp is in RFC3339 format: `yyyy-MM-ddTHH:mm:ssZ`.",
+          "type": "string",
+          "example": "2021-09-07T07:55:41Z"
+        },
+        "validUntil": {
+          "format": "date-time",
+          "description": "The timestamp from when the room can no longer be joined. The timestamp is in RFC3339 format: `yyyy-MM-ddTHH:mm:ssZ`.",
+          "type": "string",
+          "example": "2020-09-08T15:55:41Z"
+        },
+        "roomOpen": {
+          "description": "Flag to specify if the room is to be an open room or closed room.",
+          "type": "boolean",
+          "example": false
+        },
+        "participants": {
+          "description": "(Optional) Collection of identities invited to the room.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RoomParticipant"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "RoomParticipant": {
+      "description": "A participant of the room.",
+      "required": [
+        "communicationIdentifier"
+      ],
+      "type": "object",
+      "properties": {
+        "communicationIdentifier": {
+          "$ref": "#/definitions/CommunicationIdentifierModel"
+        },
+        "role": {
+          "description": "Role Name.",
+          "type": "string"
```

Comment:
This is essentially a stringly-typed enum, right? What are the permissible values?

Can you use an extensible enum here? https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#use-extensible-enums

Reply:
Addressed: changed role from string to extensible enum.